core/src/main/java/com/gorisse/thomas/sceneform/arcore/LightEstimate.kt

```diff
@@ -0,0 +1,263 @@
+            val faceOffsets = IntArray(arImages.size)
+            val buffer = Texture.PixelBufferDescriptor(
+                ByteBuffer.allocateDirect(
+                    width * height *
+                            arImages.size *
+                            // RGB Bytes per pixel
+                            6 * 2
+                ).apply {
+                    // Use the device hardware's native byte order
+                    order(ByteOrder.nativeOrder())
+
+                    val rgbaBytes = ByteArray(8) // ARGB Bytes per pixel
+                    arImages.forEachIndexed { index, image ->
+                        faceOffsets[index] = position()
+                        image.planes[0].buffer.let { imageBuffer ->
+                            while (imageBuffer.hasRemaining()) {
+                                // Only take the RGB channels
+                                put(rgbaBytes.apply {
+                                    imageBuffer.get(this)
+                                } // Skip the Alpha channel
+                                    .sliceArray(0..5))
+                            }
+                        }
+                        image.close()
+                    }
+                    flip()
+                },
+                Texture.Format.RGB,
+                Texture.Type.HALF
+            )
+
+            // Reuse the previous texture instead of creating a new one for performance and memory
+            val texture = previousEstimate?.reflections?.takeIf {
+                it.getWidth(0) == width && it.getHeight(0) == height
+            } ?: Texture.Builder()
+                .width(width)
+                .height(height)
+                .levels(0xff)
+                .sampler(Texture.Sampler.SAMPLER_CUBEMAP)
+                .format(Texture.InternalFormat.R11F_G11F_B10F)
+                .build(Filament.engine)
+
+            texture.apply {
+                if (withSpecularFilter) {
+                    generatePrefilterMipmap(Filament.engine,
+                        buffer,
+                        faceOffsets,
+                        Texture.PrefilterOptions().apply {
+                            mirror = false
+                        })
+                } else {
+                    setImage(Filament.engine, 0, buffer, faceOffsets)
+                }
+            }
+        }
+    } else {
+        baseEnvironment?.indirectLight?.reflectionsTexture
+    },
+    irradiance = if (withIrradiance) {
+        environmentalHdrAmbientSphericalHarmonics?.let { sphericalHarmonics ->
+            sphericalHarmonics.mapIndexed { index, sphericalHarmonic ->
+                sphericalHarmonic *
+                        // Convert Environmental HDR's spherical harmonics to Filament
+                        // irradiance spherical harmonics.
+                        // TODO: Should we apply the cameraExposureFactor?
+                        Environment.SPHERICAL_HARMONICS_IRRADIANCE_FACTORS[index / 3]
```

Comment:
@romainguy I hardly tried to understand what are spherical harmonics are made of without result.
Do we have to apply the camera scale factor here?